self_test/generic/complex_matrices_test/cr_complex_matrix_test.cc

```diff
@@ -0,0 +1,252 @@
+  x[1] = complex<double>(-2.0, 3.0);
+  x[2] = complex<double>(0.5, 2.2);
+  x[3] = complex<double>(-1.4, 3.0);
+
+  Vector<std::complex<double>> soln(vector_length);
+
+  // test multiply
+  matrix_square.multiply(x, soln);
+  print_complex_vector(soln);
+
+  // test multiply transposed
+  matrix_square.multiply_transpose(x, soln);
+  print_complex_vector(soln);
+
+  LinearAlgebraDistribution* dist_pt = new LinearAlgebraDistribution();
+
+  constexpr unsigned n_nz_double = 5;
+
+  Vector<double> value_double(5);
+  value_double[0] = double(1.0);
+  value_double[1] = double(2.0);
+  value_double[2] = double(3.0);
+  value_double[3] = double(4.0);
+  value_double[4] = double(5.0);
+
+  Vector<int> column_index_double(n_nz_double);
+  column_index_double[0] = 0;
+  column_index_double[1] = 2;
+  column_index_double[2] = 0;
+  column_index_double[3] = 1;
+  column_index_double[4] = 1;
+
+  Vector<int> row_start_double(n_col_square + 1);
+  row_start_double[0] = 0;
+  row_start_double[1] = 2;
+  row_start_double[2] = 3;
+  row_start_double[3] = 4;
+  row_start_double[4] = n_nz_double;
+
+  CRDoubleMatrix matrix_in_double(
+    dist_pt, n_col_square, value_double, column_index_double, row_start_double);
+
+  // test add (CRDoubleMatrix)
+  matrix_square.add(matrix_in_double, matrix_square);
+  print_cr_complex_matrix(matrix_square);
+
+  // test add (CRComplexMatrix)
+  matrix_square.add(matrix_square, matrix_square);
+  print_cr_complex_matrix(matrix_square);
+
+  // Print input matrices
+  cout << "A" << endl;
+  print_cr_complex_matrix(matrix_square);
+
+  CRComplexMatrix matrix_square_2(values_square,
+                                column_index_square,
+                                row_start_square,
+                                n_col_square,
+                                n_row_square);
+
+  cout << "B" << endl;
+  print_cr_complex_matrix(matrix_square_2);
+
+  CRComplexMatrix matrix_result(values_square,
+                                column_index_square,
+                                row_start_square,
+                                n_col_square,
+                                n_row_square);
+
+  // test multiply method 2 (default)
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // test multiply set method and default value
+  unsigned* method_pt = new unsigned;
+  method_pt = &matrix_square.serial_matrix_matrix_multiply_method();
+  cout << *method_pt << endl;
+
+  // test multiply method 1
+  *method_pt = 1;
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << *method_pt << endl;
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // test multiply method 3
+  *method_pt = 3;
+  matrix_square.multiply(matrix_square_2, matrix_result);
+  cout << *method_pt << endl;
+  cout << "A B" << endl;
+  print_cr_complex_matrix(matrix_result);
+
+  // Print cleanup messages for easier debugging
```

Comment:
Assuming that's the bit Puneet was commenting on, this is now OK. Or are you saying that this is still not working?

Reply:
Just to clarify that my comment referred to the code below line 243, not what's actually displayed here.

Reply:
It is not currently working.
Go to my oomph-lib fork on github, then actions along the top and you should see the different commits being tested. (You might already know that but just to be verbose)

Reply:
See https://github.com/oomph-lib/oomph-lib/pull/48#discussion_r718746492.

Reply:
Also, it can be helpful to just paste a link to each Action or the Actions page, rather than us having to search it out ourselves.

Reply:
Teach a man to fish and all that...
Here's the link: https://github.com/PaulAlexander23/oomph-lib/actions

Reply:
Great! Looks like the pesky macOS tests aren't working though... 

I know you've already been waiting a billion years, but I think I have a fix for that and I'm going to create a PR to get the changes in. Is there any chance you could wait until the changes have been merged in the pull them into your branch so that we can make sure either the tests all pass or that we can actually identify the failing self-test(s)?

Reply:
Sure. Let me know when you are ready for me to pull in the new changes.

Reply:
Excellent, the changes have been merged in. Can you rebase/merge to see if the self-tests complete now? Thanks!